src/viam/sessions_client.py

```diff
@@ -103,66 +109,58 @@ async def metadata(self) -> _MetadataLike:
         if response.heartbeat_window is None:
             raise GRPCError(status=Status.INTERNAL, message="Expected heartbeat window in response to start session")
 
-        self._supported.value = _SupportedState.TRUE
-        self._heartbeat_interval = response.heartbeat_window.ToTimedelta()
-        self._current_id = response.id
+        with self._lock:
+            self._supported = _SupportedState.TRUE
+            self._heartbeat_interval = response.heartbeat_window.ToTimedelta()
+            self._current_id = response.id
 
         # tick once to ensure heartbeats are supported
-        await _heartbeat_tick(self.client, self._current_id, self._supported)
+        await self._heartbeat_tick(self.client)
+
+        with self._lock:
+            if self._supported == _SupportedState.TRUE:
+                # We send heartbeats slightly faster than the interval window to
+                # ensure that we don't fall outside of it and expire the session.
```

Comment:
completely unrelated and my bad, but 1/5th the time is not "slightly" faster :clown_face: 

```suggestion
                # We send heartbeats faster than the interval window to
                # ensure that we don't fall outside of it and expire the session.
```

alternatively we can just remove the comment altogether